src/test/java/org/takes/rs/RsPrintTest.java

```diff
@@ -43,4 +51,54 @@ public void failsOnInvalidHeader() throws IOException {
         new RsPrint(new RsWithHeader("name", "\n\n\n")).print();
     }
 
+    /**
+     * RsPrint can flush body contents even when exception happens.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void flushBodyEvenWhenExceptionHappens() throws IOException {
+        final OutputStream output = Mockito.mock(OutputStream.class);
+        final IOException exception = new IOException("Error");
+        Mockito.doThrow(exception).when(output)
```

Comment:
@t-izbassar I'd not use "Mockito" where it's possible, because the test becomes more complex: it is 9 lines of mockito setup and 8 lines of actual test code. I think it would be better to define fake `OutputStream` implementation which will throw an exception on each `write()` call. See [this blog post](http://www.yegor256.com/2014/09/23/built-in-fake-objects.html) also.